### PR TITLE
[1LP][RFR] Remove uses of uncollectif where provider markers should be used

### PR DESCRIPTION
**Pull request by mshriver**
Trying to keep test collection consistent here, but not looking to fix any tests that may be failing.

Removed some old BZ's and appliance version references as I came across them

Added an is_downstream property to DummyAppliance

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 96830a700a73ace2ce6360bbb50337dbe99ffe14

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`azone`** is used in the following files:
- **cfme/tests/candu/test_azone_graph.py**
    - *test_azone_graph_screen*

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_host_graph.py**
    - *test_host_most_recent_hour_graph_screen*
    - *test_host_graph_screen*

The *local* fixture **`new_vm`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`all_possible_roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *roles*

The *local* fixture **`roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *test_server_roles_changing*

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`metering_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_metering_report.py**
    - *test_validate_cpu_usage*
    - *test_validate_memory_usage*
    - *test_validate_network_usage*
    - *test_validate_disk_usage*
    - *test_validate_storage_usage*

The *local* fixture **`new_compute_rate`** is used in the following files:
- **cfme/tests/intelligence/reports/test_validate_chargeback_report.py**
    - *assign_custom_rate*
    - *chargeback_costs_custom*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit c9f6d8d7973676163d6ce4c6cf830cdb0d06fcad

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`azone`** is used in the following files:
- **cfme/tests/candu/test_azone_graph.py**
    - *test_azone_graph_screen*

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_host_graph.py**
    - *test_host_most_recent_hour_graph_screen*
    - *test_host_graph_screen*

The *local* fixture **`new_vm`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`all_possible_roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *roles*

The *local* fixture **`roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *test_server_roles_changing*

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`metering_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_metering_report.py**
    - *test_validate_cpu_usage*
    - *test_validate_memory_usage*
    - *test_validate_network_usage*
    - *test_validate_disk_usage*
    - *test_validate_storage_usage*

The *local* fixture **`new_compute_rate`** is used in the following files:
- **cfme/tests/intelligence/reports/test_validate_chargeback_report.py**
    - *assign_custom_rate*
    - *chargeback_costs_custom*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 781da6184dcaf0c5e809d6ad6cf8dd924ad68088

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`azone`** is used in the following files:
- **cfme/tests/candu/test_azone_graph.py**
    - *test_azone_graph_screen*

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_host_graph.py**
    - *test_host_most_recent_hour_graph_screen*
    - *test_host_graph_screen*

The *local* fixture **`new_vm`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`all_possible_roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *roles*

The *local* fixture **`roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *test_server_roles_changing*

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`metering_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_metering_report.py**
    - *test_validate_cpu_usage*
    - *test_validate_memory_usage*
    - *test_validate_network_usage*
    - *test_validate_disk_usage*
    - *test_validate_storage_usage*

The *local* fixture **`new_compute_rate`** is used in the following files:
- **cfme/tests/intelligence/reports/test_validate_chargeback_report.py**
    - *assign_custom_rate*
    - *chargeback_costs_custom*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 11c00617177428a0c0cd5825c8aeda20965f57dd

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`azone`** is used in the following files:
- **cfme/tests/candu/test_azone_graph.py**
    - *test_azone_graph_screen*

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_host_graph.py**
    - *test_host_most_recent_hour_graph_screen*
    - *test_host_graph_screen*

The *local* fixture **`new_vm`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`all_possible_roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *roles*

The *local* fixture **`roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *test_server_roles_changing*

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`metering_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_metering_report.py**
    - *test_validate_cpu_usage*
    - *test_validate_memory_usage*
    - *test_validate_network_usage*
    - *test_validate_disk_usage*
    - *test_validate_storage_usage*

The *local* fixture **`new_compute_rate`** is used in the following files:
- **cfme/tests/intelligence/reports/test_validate_chargeback_report.py**
    - *assign_custom_rate*
    - *chargeback_costs_custom*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 954c44f62518722b55437e1dde4c6a07874b4b25

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`azone`** is used in the following files:
- **cfme/tests/candu/test_azone_graph.py**
    - *test_azone_graph_screen*

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_host_graph.py**
    - *test_host_most_recent_hour_graph_screen*
    - *test_host_graph_screen*

The *local* fixture **`new_vm`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`all_possible_roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *roles*

The *local* fixture **`roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *test_server_roles_changing*

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`metering_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_metering_report.py**
    - *test_validate_cpu_usage*
    - *test_validate_memory_usage*
    - *test_validate_network_usage*
    - *test_validate_disk_usage*
    - *test_validate_storage_usage*

The *local* fixture **`new_compute_rate`** is used in the following files:
- **cfme/tests/intelligence/reports/test_validate_chargeback_report.py**
    - *assign_custom_rate*
    - *chargeback_costs_custom*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit efc4b15c977bc3c97d1b1ab5161080c9cb2faa4c

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`azone`** is used in the following files:
- **cfme/tests/candu/test_azone_graph.py**
    - *test_azone_graph_screen*

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_host_graph.py**
    - *test_host_most_recent_hour_graph_screen*
    - *test_host_graph_screen*

The *local* fixture **`new_vm`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`all_possible_roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *roles*

The *local* fixture **`roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *test_server_roles_changing*

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`metering_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_metering_report.py**
    - *test_validate_cpu_usage*
    - *test_validate_memory_usage*
    - *test_validate_network_usage*
    - *test_validate_disk_usage*
    - *test_validate_storage_usage*

The *local* fixture **`new_compute_rate`** is used in the following files:
- **cfme/tests/intelligence/reports/test_validate_chargeback_report.py**
    - *assign_custom_rate*
    - *chargeback_costs_custom*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/cloud_infra_common/test_snapshots_rest.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def test_delete_snapshot_race(self, request, appliance, collection, vm):
                 raise
 
     @pytest.mark.rhv2
-    @pytest.mark.uncollectif(lambda provider: not provider.one_of(InfraProvider))
+    @pytest.mark.provider([VMwareProvider, RHEVMProvider], override=True)
```
Comment (izapolsk) [inline cfme/tests/cloud_infra_common/test_snapshots_rest.py]: there is also SCVMM.  is it skipped somewhere ?

**Comment by izapolsk** [comment]
PRT isn't ready yet

File: cfme/tests/control/test_alerts.py [hunk 2]
```diff
@@ -26,15 +26,14 @@
 from cfme.utils.wait import wait_for
 
 pf1 = ProviderFilter(classes=[InfraProvider])
-pf2 = ProviderFilter(classes=[SCVMMProvider], inverted=True)
+pf2 = ProviderFilter(classes=[SCVMMProvider, RHEVMProvider, KubeVirtProvider], inverted=True)
```
[the remark this answers is not in the document]
Comment (mshriver) [reply cfme/tests/control/test_alerts.py]: Yeah you're right, I can just drop it from this.

File: cfme/tests/cloud_infra_common/test_snapshots_rest.py (hunk 1 above)
Comment (mshriver) [reply cfme/tests/cloud_infra_common/test_snapshots_rest.py]: Module marker does not include SCVMM, it specifically includes Openstack, Rhevm, VMware (and then the uncollect was uncollecting openstack). So Just RHEVM and VMWare here, though SCMM might be relevant. I'll leave that up to snapshots owner to add that coverage.

File: cfme/tests/control/test_alerts.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mshriver) [reply cfme/tests/control/test_alerts.py]: helps if I actually push the commit.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 0ada9238006879d2a583dc799f3c5f640caee4c5

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`azone`** is used in the following files:
- **cfme/tests/candu/test_azone_graph.py**
    - *test_azone_graph_screen*

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_host_graph.py**
    - *test_host_most_recent_hour_graph_screen*
    - *test_host_graph_screen*

The *local* fixture **`new_vm`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_cockpit_server.py**
    - *test_cockpit_server_role*

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`all_possible_roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *roles*

The *local* fixture **`roles`** is used in the following files:
- **cfme/tests/configure/test_server_roles.py**
    - *test_server_roles_changing*

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`metering_report`** is used in the following files:
- **cfme/tests/intelligence/reports/test_metering_report.py**
    - *test_validate_cpu_usage*
    - *test_validate_memory_usage*
    - *test_validate_network_usage*
    - *test_validate_disk_usage*
    - *test_validate_storage_usage*

The *local* fixture **`new_compute_rate`** is used in the following files:
- **cfme/tests/intelligence/reports/test_validate_chargeback_report.py**
    - *assign_custom_rate*
    - *chargeback_costs_custom*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**